// simulation_e/npm.js


var products = [
  {
    id : 1,
    product : 'Computer',
    quantity : 4
  },
  {
    id : 2,
    product : 'Phone',
    quantity : 4
  },
  {
    id : 3,
    product : 'Television',
    quantity : 4
  },
  {
    id : 4,
    product : 'Mp3 player',
    quantity : 4
  },
  {
    id : 5,
    product : 'CD player',
    quantity : 4
  }
];

var users = [
  {
    token : 'Admin'
  },
  {
    token : 'Pippo'
  },
  {
    token : 'Caio'
  },
  {
    token : 'Sempronio'
  }
];

exports.allUsers = function(){
  return users;
}

exports.allProducts = function(){
  return products;
}


exports.insertProduct = function(token,product){
  if(token == 'Admin'){

    products.push({id : products[products.length-1].id+1,product: product, quantity : 1} )
    return products;

  }else{
    return 'can not insert the product';
    }
  }





  exports.deleteProduct = function(token,id){
    if(token == 'Admin'){
      for (var i = 0; i < products.length; i++) {
        if(id == products[i].id){
          products.splice(i,1)
        }
      }
      return products;
        }else{
        return 'can not delete the product';
        }
      }




  exports.modifyProduct = function(token,id,newProduct,quantity){
    if(token == 'Admin'){
      for (var i = 0; i < products.length; i++) {
        if(id == products[i].id){
          products[i].product = newProduct;
          products[i].quantity = quantity;
          }
        }
      return products;
    }
  }





exports.buyProduct = function(token,id){
  var temp = {Message: 'Error, token or id uncorrected'};
      for (var i = 0; i < users.length; i++) {
        for (var x = 0; x < products.length; x++) {
          if (token == users[i].token &&  id == products[x].id && token != 'Admin') {
              temp = {token : users[i].token, product : products[x].product , id : products[x].id, time : new Date()};
          }

        }
      }
     return temp
};
